<?xml version="1.0" encoding="UTF-8"?>
<testsuite xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://maven.apache.org/surefire/maven-surefire-plugin/xsd/surefire-test-report-3.0.xsd" version="3.0" name="code.restful.AddressBookControllerTest" time="4.747" tests="9" errors="0" skipped="0" failures="9">
  <properties>
    <property name="java.specification.version" value="21"/>
    <property name="sun.cpu.isalist" value="amd64"/>
    <property name="sun.jnu.encoding" value="Cp1252"/>
    <property name="java.class.path" value="C:\Cursor\addressbook\addressbook-rest-service\target\test-classes;C:\Cursor\addressbook\addressbook-rest-service\target\classes;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter-web\3.2.3\spring-boot-starter-web-3.2.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter\3.2.3\spring-boot-starter-3.2.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot\3.2.3\spring-boot-3.2.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-autoconfigure\3.2.3\spring-boot-autoconfigure-3.2.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter-logging\3.2.3\spring-boot-starter-logging-3.2.3.jar;C:\Users\navee\.m2\repository\ch\qos\logback\logback-classic\1.4.14\logback-classic-1.4.14.jar;C:\Users\navee\.m2\repository\ch\qos\logback\logback-core\1.4.14\logback-core-1.4.14.jar;C:\Users\navee\.m2\repository\org\apache\logging\log4j\log4j-to-slf4j\2.21.1\log4j-to-slf4j-2.21.1.jar;C:\Users\navee\.m2\repository\org\apache\logging\log4j\log4j-api\2.21.1\log4j-api-2.21.1.jar;C:\Users\navee\.m2\repository\org\slf4j\jul-to-slf4j\2.0.12\jul-to-slf4j-2.0.12.jar;C:\Users\navee\.m2\repository\org\yaml\snakeyaml\2.2\snakeyaml-2.2.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter-json\3.2.3\spring-boot-starter-json-3.2.3.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\core\jackson-databind\2.15.4\jackson-databind-2.15.4.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\core\jackson-annotations\2.15.4\jackson-annotations-2.15.4.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\core\jackson-core\2.15.4\jackson-core-2.15.4.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\datatype\jackson-datatype-jdk8\2.15.4\jackson-datatype-jdk8-2.15.4.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\datatype\jackson-datatype-jsr310\2.15.4\jackson-datatype-jsr310-2.15.4.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\module\jackson-module-parameter-names\2.15.4\jackson-module-parameter-names-2.15.4.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter-tomcat\3.2.3\spring-boot-starter-tomcat-3.2.3.jar;C:\Users\navee\.m2\repository\org\apache\tomcat\embed\tomcat-embed-core\10.1.19\tomcat-embed-core-10.1.19.jar;C:\Users\navee\.m2\repository\org\apache\tomcat\embed\tomcat-embed-websocket\10.1.19\tomcat-embed-websocket-10.1.19.jar;C:\Users\navee\.m2\repository\org\springframework\spring-web\6.1.4\spring-web-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\spring-beans\6.1.4\spring-beans-6.1.4.jar;C:\Users\navee\.m2\repository\io\micrometer\micrometer-observation\1.12.3\micrometer-observation-1.12.3.jar;C:\Users\navee\.m2\repository\io\micrometer\micrometer-commons\1.12.3\micrometer-commons-1.12.3.jar;C:\Users\navee\.m2\repository\org\springframework\spring-webmvc\6.1.4\spring-webmvc-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\spring-aop\6.1.4\spring-aop-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\spring-context\6.1.4\spring-context-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\spring-expression\6.1.4\spring-expression-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter-validation\3.2.3\spring-boot-starter-validation-3.2.3.jar;C:\Users\navee\.m2\repository\org\apache\tomcat\embed\tomcat-embed-el\10.1.19\tomcat-embed-el-10.1.19.jar;C:\Users\navee\.m2\repository\org\hibernate\validator\hibernate-validator\8.0.1.Final\hibernate-validator-8.0.1.Final.jar;C:\Users\navee\.m2\repository\jakarta\validation\jakarta.validation-api\3.0.2\jakarta.validation-api-3.0.2.jar;C:\Users\navee\.m2\repository\org\jboss\logging\jboss-logging\3.5.3.Final\jboss-logging-3.5.3.Final.jar;C:\Users\navee\.m2\repository\com\fasterxml\classmate\1.6.0\classmate-1.6.0.jar;C:\Users\navee\.m2\repository\org\springdoc\springdoc-openapi-starter-webmvc-ui\2.3.0\springdoc-openapi-starter-webmvc-ui-2.3.0.jar;C:\Users\navee\.m2\repository\org\springdoc\springdoc-openapi-starter-webmvc-api\2.3.0\springdoc-openapi-starter-webmvc-api-2.3.0.jar;C:\Users\navee\.m2\repository\org\springdoc\springdoc-openapi-starter-common\2.3.0\springdoc-openapi-starter-common-2.3.0.jar;C:\Users\navee\.m2\repository\io\swagger\core\v3\swagger-core-jakarta\2.2.19\swagger-core-jakarta-2.2.19.jar;C:\Users\navee\.m2\repository\org\apache\commons\commons-lang3\3.13.0\commons-lang3-3.13.0.jar;C:\Users\navee\.m2\repository\io\swagger\core\v3\swagger-annotations-jakarta\2.2.19\swagger-annotations-jakarta-2.2.19.jar;C:\Users\navee\.m2\repository\io\swagger\core\v3\swagger-models-jakarta\2.2.19\swagger-models-jakarta-2.2.19.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\dataformat\jackson-dataformat-yaml\2.15.4\jackson-dataformat-yaml-2.15.4.jar;C:\Users\navee\.m2\repository\org\webjars\swagger-ui\5.10.3\swagger-ui-5.10.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter-test\3.2.3\spring-boot-starter-test-3.2.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-test\3.2.3\spring-boot-test-3.2.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-test-autoconfigure\3.2.3\spring-boot-test-autoconfigure-3.2.3.jar;C:\Users\navee\.m2\repository\com\jayway\jsonpath\json-path\2.9.0\json-path-2.9.0.jar;C:\Users\navee\.m2\repository\org\slf4j\slf4j-api\2.0.12\slf4j-api-2.0.12.jar;C:\Users\navee\.m2\repository\jakarta\xml\bind\jakarta.xml.bind-api\4.0.1\jakarta.xml.bind-api-4.0.1.jar;C:\Users\navee\.m2\repository\jakarta\activation\jakarta.activation-api\2.1.2\jakarta.activation-api-2.1.2.jar;C:\Users\navee\.m2\repository\net\minidev\json-smart\2.5.0\json-smart-2.5.0.jar;C:\Users\navee\.m2\repository\net\minidev\accessors-smart\2.5.0\accessors-smart-2.5.0.jar;C:\Users\navee\.m2\repository\org\ow2\asm\asm\9.3\asm-9.3.jar;C:\Users\navee\.m2\repository\org\assertj\assertj-core\3.24.2\assertj-core-3.24.2.jar;C:\Users\navee\.m2\repository\net\bytebuddy\byte-buddy\1.14.12\byte-buddy-1.14.12.jar;C:\Users\navee\.m2\repository\org\awaitility\awaitility\4.2.0\awaitility-4.2.0.jar;C:\Users\navee\.m2\repository\org\hamcrest\hamcrest\2.2\hamcrest-2.2.jar;C:\Users\navee\.m2\repository\org\junit\jupiter\junit-jupiter\5.10.2\junit-jupiter-5.10.2.jar;C:\Users\navee\.m2\repository\org\junit\jupiter\junit-jupiter-params\5.10.2\junit-jupiter-params-5.10.2.jar;C:\Users\navee\.m2\repository\org\mockito\mockito-core\5.7.0\mockito-core-5.7.0.jar;C:\Users\navee\.m2\repository\net\bytebuddy\byte-buddy-agent\1.14.12\byte-buddy-agent-1.14.12.jar;C:\Users\navee\.m2\repository\org\objenesis\objenesis\3.3\objenesis-3.3.jar;C:\Users\navee\.m2\repository\org\mockito\mockito-junit-jupiter\5.7.0\mockito-junit-jupiter-5.7.0.jar;C:\Users\navee\.m2\repository\org\skyscreamer\jsonassert\1.5.1\jsonassert-1.5.1.jar;C:\Users\navee\.m2\repository\com\vaadin\external\google\android-json\0.0.20131108.vaadin1\android-json-0.0.20131108.vaadin1.jar;C:\Users\navee\.m2\repository\org\springframework\spring-core\6.1.4\spring-core-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\spring-jcl\6.1.4\spring-jcl-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\spring-test\6.1.4\spring-test-6.1.4.jar;C:\Users\navee\.m2\repository\org\xmlunit\xmlunit-core\2.9.1\xmlunit-core-2.9.1.jar;C:\Users\navee\.m2\repository\org\junit\jupiter\junit-jupiter-api\5.10.1\junit-jupiter-api-5.10.1.jar;C:\Users\navee\.m2\repository\org\opentest4j\opentest4j\1.3.0\opentest4j-1.3.0.jar;C:\Users\navee\.m2\repository\org\junit\platform\junit-platform-commons\1.10.2\junit-platform-commons-1.10.2.jar;C:\Users\navee\.m2\repository\org\apiguardian\apiguardian-api\1.1.2\apiguardian-api-1.1.2.jar;C:\Users\navee\.m2\repository\org\junit\jupiter\junit-jupiter-engine\5.10.1\junit-jupiter-engine-5.10.1.jar;C:\Users\navee\.m2\repository\org\junit\platform\junit-platform-engine\1.10.2\junit-platform-engine-1.10.2.jar;C:\Users\navee\.m2\repository\org\projectlombok\lombok\1.18.30\lombok-1.18.30.jar;C:\Users\navee\.m2\repository\jakarta\annotation\jakarta.annotation-api\2.1.1\jakarta.annotation-api-2.1.1.jar;C:\Users\navee\.m2\repository\jakarta\persistence\jakarta.persistence-api\3.1.0\jakarta.persistence-api-3.1.0.jar;"/>
    <property name="java.vm.vendor" value="Eclipse Adoptium"/>
    <property name="sun.arch.data.model" value="64"/>
    <property name="user.variant" value=""/>
    <property name="java.vendor.url" value="https://adoptium.net/"/>
    <property name="user.timezone" value="Australia/Sydney"/>
    <property name="org.jboss.logging.provider" value="slf4j"/>
    <property name="os.name" value="Windows 11"/>
    <property name="java.vm.specification.version" value="21"/>
    <property name="sun.java.launcher" value="SUN_STANDARD"/>
    <property name="user.country" value="AU"/>
    <property name="sun.boot.library.path" value="C:\Program Files\Eclipse Adoptium\jdk-21.0.7.6-hotspot\bin"/>
    <property name="sun.java.command" value="C:\Users\navee\AppData\Local\Temp\surefire9045917560496583140\surefirebooter-20250618154528394_3.jar C:\Users\navee\AppData\Local\Temp\surefire9045917560496583140 2025-06-18T15-45-28_251-jvmRun1 surefire-20250618154528394_1tmp surefire_0-20250618154528394_2tmp"/>
    <property name="jdk.debug" value="release"/>
    <property name="surefire.test.class.path" value="C:\Cursor\addressbook\addressbook-rest-service\target\test-classes;C:\Cursor\addressbook\addressbook-rest-service\target\classes;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter-web\3.2.3\spring-boot-starter-web-3.2.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter\3.2.3\spring-boot-starter-3.2.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot\3.2.3\spring-boot-3.2.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-autoconfigure\3.2.3\spring-boot-autoconfigure-3.2.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter-logging\3.2.3\spring-boot-starter-logging-3.2.3.jar;C:\Users\navee\.m2\repository\ch\qos\logback\logback-classic\1.4.14\logback-classic-1.4.14.jar;C:\Users\navee\.m2\repository\ch\qos\logback\logback-core\1.4.14\logback-core-1.4.14.jar;C:\Users\navee\.m2\repository\org\apache\logging\log4j\log4j-to-slf4j\2.21.1\log4j-to-slf4j-2.21.1.jar;C:\Users\navee\.m2\repository\org\apache\logging\log4j\log4j-api\2.21.1\log4j-api-2.21.1.jar;C:\Users\navee\.m2\repository\org\slf4j\jul-to-slf4j\2.0.12\jul-to-slf4j-2.0.12.jar;C:\Users\navee\.m2\repository\org\yaml\snakeyaml\2.2\snakeyaml-2.2.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter-json\3.2.3\spring-boot-starter-json-3.2.3.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\core\jackson-databind\2.15.4\jackson-databind-2.15.4.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\core\jackson-annotations\2.15.4\jackson-annotations-2.15.4.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\core\jackson-core\2.15.4\jackson-core-2.15.4.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\datatype\jackson-datatype-jdk8\2.15.4\jackson-datatype-jdk8-2.15.4.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\datatype\jackson-datatype-jsr310\2.15.4\jackson-datatype-jsr310-2.15.4.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\module\jackson-module-parameter-names\2.15.4\jackson-module-parameter-names-2.15.4.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter-tomcat\3.2.3\spring-boot-starter-tomcat-3.2.3.jar;C:\Users\navee\.m2\repository\org\apache\tomcat\embed\tomcat-embed-core\10.1.19\tomcat-embed-core-10.1.19.jar;C:\Users\navee\.m2\repository\org\apache\tomcat\embed\tomcat-embed-websocket\10.1.19\tomcat-embed-websocket-10.1.19.jar;C:\Users\navee\.m2\repository\org\springframework\spring-web\6.1.4\spring-web-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\spring-beans\6.1.4\spring-beans-6.1.4.jar;C:\Users\navee\.m2\repository\io\micrometer\micrometer-observation\1.12.3\micrometer-observation-1.12.3.jar;C:\Users\navee\.m2\repository\io\micrometer\micrometer-commons\1.12.3\micrometer-commons-1.12.3.jar;C:\Users\navee\.m2\repository\org\springframework\spring-webmvc\6.1.4\spring-webmvc-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\spring-aop\6.1.4\spring-aop-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\spring-context\6.1.4\spring-context-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\spring-expression\6.1.4\spring-expression-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter-validation\3.2.3\spring-boot-starter-validation-3.2.3.jar;C:\Users\navee\.m2\repository\org\apache\tomcat\embed\tomcat-embed-el\10.1.19\tomcat-embed-el-10.1.19.jar;C:\Users\navee\.m2\repository\org\hibernate\validator\hibernate-validator\8.0.1.Final\hibernate-validator-8.0.1.Final.jar;C:\Users\navee\.m2\repository\jakarta\validation\jakarta.validation-api\3.0.2\jakarta.validation-api-3.0.2.jar;C:\Users\navee\.m2\repository\org\jboss\logging\jboss-logging\3.5.3.Final\jboss-logging-3.5.3.Final.jar;C:\Users\navee\.m2\repository\com\fasterxml\classmate\1.6.0\classmate-1.6.0.jar;C:\Users\navee\.m2\repository\org\springdoc\springdoc-openapi-starter-webmvc-ui\2.3.0\springdoc-openapi-starter-webmvc-ui-2.3.0.jar;C:\Users\navee\.m2\repository\org\springdoc\springdoc-openapi-starter-webmvc-api\2.3.0\springdoc-openapi-starter-webmvc-api-2.3.0.jar;C:\Users\navee\.m2\repository\org\springdoc\springdoc-openapi-starter-common\2.3.0\springdoc-openapi-starter-common-2.3.0.jar;C:\Users\navee\.m2\repository\io\swagger\core\v3\swagger-core-jakarta\2.2.19\swagger-core-jakarta-2.2.19.jar;C:\Users\navee\.m2\repository\org\apache\commons\commons-lang3\3.13.0\commons-lang3-3.13.0.jar;C:\Users\navee\.m2\repository\io\swagger\core\v3\swagger-annotations-jakarta\2.2.19\swagger-annotations-jakarta-2.2.19.jar;C:\Users\navee\.m2\repository\io\swagger\core\v3\swagger-models-jakarta\2.2.19\swagger-models-jakarta-2.2.19.jar;C:\Users\navee\.m2\repository\com\fasterxml\jackson\dataformat\jackson-dataformat-yaml\2.15.4\jackson-dataformat-yaml-2.15.4.jar;C:\Users\navee\.m2\repository\org\webjars\swagger-ui\5.10.3\swagger-ui-5.10.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-starter-test\3.2.3\spring-boot-starter-test-3.2.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-test\3.2.3\spring-boot-test-3.2.3.jar;C:\Users\navee\.m2\repository\org\springframework\boot\spring-boot-test-autoconfigure\3.2.3\spring-boot-test-autoconfigure-3.2.3.jar;C:\Users\navee\.m2\repository\com\jayway\jsonpath\json-path\2.9.0\json-path-2.9.0.jar;C:\Users\navee\.m2\repository\org\slf4j\slf4j-api\2.0.12\slf4j-api-2.0.12.jar;C:\Users\navee\.m2\repository\jakarta\xml\bind\jakarta.xml.bind-api\4.0.1\jakarta.xml.bind-api-4.0.1.jar;C:\Users\navee\.m2\repository\jakarta\activation\jakarta.activation-api\2.1.2\jakarta.activation-api-2.1.2.jar;C:\Users\navee\.m2\repository\net\minidev\json-smart\2.5.0\json-smart-2.5.0.jar;C:\Users\navee\.m2\repository\net\minidev\accessors-smart\2.5.0\accessors-smart-2.5.0.jar;C:\Users\navee\.m2\repository\org\ow2\asm\asm\9.3\asm-9.3.jar;C:\Users\navee\.m2\repository\org\assertj\assertj-core\3.24.2\assertj-core-3.24.2.jar;C:\Users\navee\.m2\repository\net\bytebuddy\byte-buddy\1.14.12\byte-buddy-1.14.12.jar;C:\Users\navee\.m2\repository\org\awaitility\awaitility\4.2.0\awaitility-4.2.0.jar;C:\Users\navee\.m2\repository\org\hamcrest\hamcrest\2.2\hamcrest-2.2.jar;C:\Users\navee\.m2\repository\org\junit\jupiter\junit-jupiter\5.10.2\junit-jupiter-5.10.2.jar;C:\Users\navee\.m2\repository\org\junit\jupiter\junit-jupiter-params\5.10.2\junit-jupiter-params-5.10.2.jar;C:\Users\navee\.m2\repository\org\mockito\mockito-core\5.7.0\mockito-core-5.7.0.jar;C:\Users\navee\.m2\repository\net\bytebuddy\byte-buddy-agent\1.14.12\byte-buddy-agent-1.14.12.jar;C:\Users\navee\.m2\repository\org\objenesis\objenesis\3.3\objenesis-3.3.jar;C:\Users\navee\.m2\repository\org\mockito\mockito-junit-jupiter\5.7.0\mockito-junit-jupiter-5.7.0.jar;C:\Users\navee\.m2\repository\org\skyscreamer\jsonassert\1.5.1\jsonassert-1.5.1.jar;C:\Users\navee\.m2\repository\com\vaadin\external\google\android-json\0.0.20131108.vaadin1\android-json-0.0.20131108.vaadin1.jar;C:\Users\navee\.m2\repository\org\springframework\spring-core\6.1.4\spring-core-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\spring-jcl\6.1.4\spring-jcl-6.1.4.jar;C:\Users\navee\.m2\repository\org\springframework\spring-test\6.1.4\spring-test-6.1.4.jar;C:\Users\navee\.m2\repository\org\xmlunit\xmlunit-core\2.9.1\xmlunit-core-2.9.1.jar;C:\Users\navee\.m2\repository\org\junit\jupiter\junit-jupiter-api\5.10.1\junit-jupiter-api-5.10.1.jar;C:\Users\navee\.m2\repository\org\opentest4j\opentest4j\1.3.0\opentest4j-1.3.0.jar;C:\Users\navee\.m2\repository\org\junit\platform\junit-platform-commons\1.10.2\junit-platform-commons-1.10.2.jar;C:\Users\navee\.m2\repository\org\apiguardian\apiguardian-api\1.1.2\apiguardian-api-1.1.2.jar;C:\Users\navee\.m2\repository\org\junit\jupiter\junit-jupiter-engine\5.10.1\junit-jupiter-engine-5.10.1.jar;C:\Users\navee\.m2\repository\org\junit\platform\junit-platform-engine\1.10.2\junit-platform-engine-1.10.2.jar;C:\Users\navee\.m2\repository\org\projectlombok\lombok\1.18.30\lombok-1.18.30.jar;C:\Users\navee\.m2\repository\jakarta\annotation\jakarta.annotation-api\2.1.1\jakarta.annotation-api-2.1.1.jar;C:\Users\navee\.m2\repository\jakarta\persistence\jakarta.persistence-api\3.1.0\jakarta.persistence-api-3.1.0.jar;"/>
    <property name="sun.cpu.endian" value="little"/>
    <property name="user.home" value="C:\Users\navee"/>
    <property name="user.language" value="en"/>
    <property name="java.specification.vendor" value="Oracle Corporation"/>
    <property name="java.version.date" value="2025-04-15"/>
    <property name="java.home" value="C:\Program Files\Eclipse Adoptium\jdk-21.0.7.6-hotspot"/>
    <property name="file.separator" value="\"/>
    <property name="basedir" value="C:\Cursor\addressbook\addressbook-rest-service"/>
    <property name="java.vm.compressedOopsMode" value="Zero based"/>
    <property name="line.separator" value="&#10;"/>
    <property name="java.vm.specification.vendor" value="Oracle Corporation"/>
    <property name="java.specification.name" value="Java Platform API Specification"/>
    <property name="FILE_LOG_CHARSET" value="UTF-8"/>
    <property name="java.awt.headless" value="true"/>
    <property name="surefire.real.class.path" value="C:\Users\navee\AppData\Local\Temp\surefire9045917560496583140\surefirebooter-20250618154528394_3.jar"/>
    <property name="user.script" value=""/>
    <property name="sun.management.compiler" value="HotSpot 64-Bit Tiered Compilers"/>
    <property name="java.runtime.version" value="21.0.7+6-LTS"/>
    <property name="user.name" value="navee"/>
    <property name="stdout.encoding" value="Cp1252"/>
    <property name="path.separator" value=";"/>
    <property name="os.version" value="10.0"/>
    <property name="java.runtime.name" value="OpenJDK Runtime Environment"/>
    <property name="file.encoding" value="UTF-8"/>
    <property name="java.vm.name" value="OpenJDK 64-Bit Server VM"/>
    <property name="java.vendor.version" value="Temurin-21.0.7+6"/>
    <property name="localRepository" value="C:\Users\navee\.m2\repository"/>
    <property name="java.vendor.url.bug" value="https://github.com/adoptium/adoptium-support/issues"/>
    <property name="java.io.tmpdir" value="C:\Users\navee\AppData\Local\Temp\"/>
    <property name="java.version" value="21.0.7"/>
    <property name="user.dir" value="C:\Cursor\addressbook\addressbook-rest-service"/>
    <property name="os.arch" value="amd64"/>
    <property name="java.vm.specification.name" value="Java Virtual Machine Specification"/>
    <property name="PID" value="10816"/>
    <property name="sun.os.patch.level" value=""/>
    <property name="CONSOLE_LOG_CHARSET" value="UTF-8"/>
    <property name="native.encoding" value="Cp1252"/>
    <property name="java.library.path" value="C:\Program Files\Eclipse Adoptium\jdk-21.0.7.6-hotspot\bin;C:\WINDOWS\Sun\Java\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\Program Files\Eclipse Adoptium\jdk-21.0.7.6-hotspot\bin;C:\Program Files\Microsoft SDKs\Azure\CLI2\wbin;C:\Program Files\Common Files\Oracle\Java\javapath;C:\Program Files\Microsoft\jdk-17.0.6.10-hotspot\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Kubernetes\Minikube;C:\ProgramData\chocolatey\bin;C:\Users\navee\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\navee\.vs-kubernetes\tools\kubectl\kubectl.exe;C:\Users\navee\AppData\Local\Microsoft\WindowsApps;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Docker\Docker\resources\bin;C:\Program Files\Git\cmd;C:\ProgramData\chocolatey\lib\maven\apache-maven-3.9.9\bin;;C:\Program Files\PostgreSQL\17\bin;C:\Program Files\nodejs\;C:\Users\navee\AppData\Local\Programs\Python\Python313\Scripts\;C:\Users\navee\AppData\Local\Programs\Python\Python313\;C:\Users\navee\AppData\Local\Coursier\data\bin;C:\Users\navee\AppData\Local\Programs\Windsurf\bin;C:\Users\navee\AppData\Roaming\npm;C:\Program Files\nodejs;;C:\Program Files\nodejs;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\dotnet\;C:\Users\navee\AppData\Local\Programs\Python\Python313\Scripts\;C:\Users\navee\AppData\Local\Programs\Python\Python313\;C:\Users\navee\AppData\Local\Microsoft\WindowsApps;C:\Users\navee\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\navee\AppData\Local\Coursier\data\bin;C:\Users\navee\AppData\Local\Programs\Windsurf\bin;C:\Users\navee\AppData\Roaming\npm;C:\Users\navee\.dotnet\tools;C:\Users\navee\AppData\Local\Programs\cursor\resources\app\bin;."/>
    <property name="java.vm.info" value="mixed mode, sharing"/>
    <property name="stderr.encoding" value="Cp1252"/>
    <property name="java.vendor" value="Eclipse Adoptium"/>
    <property name="java.vm.version" value="21.0.7+6-LTS"/>
    <property name="sun.io.unicode.encoding" value="UnicodeLittle"/>
    <property name="java.class.version" value="65.0"/>
  </properties>
  <testcase name="addContact_WithValidData_ShouldCreateContact" classname="code.restful.AddressBookControllerTest" time="0.262">
    <failure message="Status expected:&lt;201&gt; but was:&lt;500&gt;" type="java.lang.AssertionError"><![CDATA[java.lang.AssertionError: Status expected:<201> but was:<500>
	at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
	at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
	at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
	at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
	at code.restful.AddressBookControllerTest.addContact_WithValidData_ShouldCreateContact(AddressBookControllerTest.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
]]></failure>
    <system-out><![CDATA[15:45:29.608 [main] INFO org.springframework.test.context.support.AnnotationConfigContextLoaderUtils -- Could not detect default configuration classes for test class [code.restful.AddressBookControllerTest]: AddressBookControllerTest does not declare any static, non-private, non-final, nested classes annotated with @Configuration.
15:45:29.781 [main] INFO org.springframework.boot.test.context.SpringBootTestContextBootstrapper -- Found @SpringBootConfiguration code.restful.AddressBookApplication for test class code.restful.AddressBookControllerTest

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                (v3.2.3)

2025-06-18T15:45:30.297+10:00  INFO 10816 --- [           main] code.restful.AddressBookControllerTest   : Starting AddressBookControllerTest using Java 21.0.7 with PID 10816 (started by navee in C:\Cursor\addressbook\addressbook-rest-service)
2025-06-18T15:45:30.297+10:00 DEBUG 10816 --- [           main] code.restful.AddressBookControllerTest   : Running with Spring Boot v3.2.3, Spring v6.1.4
2025-06-18T15:45:30.297+10:00  INFO 10816 --- [           main] code.restful.AddressBookControllerTest   : The following 1 profile is active: "test"
2025-06-18T15:45:30.306+10:00 DEBUG 10816 --- [           main] o.s.w.c.s.GenericWebApplicationContext   : Refreshing org.springframework.web.context.support.GenericWebApplicationContext@63a5d002
2025-06-18T15:45:30.456+10:00 DEBUG 10816 --- [           main] o.s.c.a.ClassPathBeanDefinitionScanner   : Identified candidate component class: file [C:\Cursor\addressbook\addressbook-rest-service\target\classes\code\restful\addressbook\AddressBookController.class]
2025-06-18T15:45:30.473+10:00 DEBUG 10816 --- [           main] o.s.c.a.ClassPathBeanDefinitionScanner   : Identified candidate component class: file [C:\Cursor\addressbook\addressbook-rest-service\target\classes\code\restful\config\OpenApiConfig.class]
2025-06-18T15:45:30.490+10:00 DEBUG 10816 --- [           main] o.s.c.a.ClassPathBeanDefinitionScanner   : Identified candidate component class: file [C:\Cursor\addressbook\addressbook-rest-service\target\classes\code\restful\exception\GlobalExceptionHandler.class]
2025-06-18T15:45:30.512+10:00 DEBUG 10816 --- [           main] o.s.c.a.ClassPathBeanDefinitionScanner   : Identified candidate component class: file [C:\Cursor\addressbook\addressbook-rest-service\target\classes\code\restful\persistence\addressbook\InMemoryAddressBook.class]
2025-06-18T15:45:30.518+10:00 DEBUG 10816 --- [           main] o.s.c.a.ClassPathBeanDefinitionScanner   : Identified candidate component class: file [C:\Cursor\addressbook\addressbook-rest-service\target\classes\code\restful\persistence\addressbook\InMemoryAddressBook2.class]
2025-06-18T15:45:33.217+10:00 DEBUG 10816 --- [           main] s.w.s.m.m.a.RequestMappingHandlerMapping : 7 mappings in 'requestMappingHandlerMapping'
2025-06-18T15:45:33.340+10:00 DEBUG 10816 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Patterns [/webjars/**, /**] in 'resourceHandlerMapping'
2025-06-18T15:45:33.410+10:00 DEBUG 10816 --- [           main] s.w.s.m.m.a.RequestMappingHandlerAdapter : ControllerAdvice beans: 0 @ModelAttribute, 0 @InitBinder, 1 RequestBodyAdvice, 1 ResponseBodyAdvice
2025-06-18T15:45:33.480+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : ControllerAdvice beans: 1 @ExceptionHandler, 1 ResponseBodyAdvice
2025-06-18T15:45:33.577+10:00  INFO 10816 --- [           main] o.s.b.t.m.w.SpringBootMockServletContext : Initializing Spring TestDispatcherServlet ''
2025-06-18T15:45:33.577+10:00  INFO 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Initializing Servlet ''
2025-06-18T15:45:33.577+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Detected AcceptHeaderLocaleResolver
2025-06-18T15:45:33.577+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Detected FixedThemeResolver
2025-06-18T15:45:33.577+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Detected org.springframework.web.servlet.view.DefaultRequestToViewNameTranslator@54489296
2025-06-18T15:45:33.577+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Detected org.springframework.web.servlet.support.SessionFlashMapManager@4e8598d9
2025-06-18T15:45:33.580+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : enableLoggingRequestDetails='false': request parameters and headers will be masked to prevent unsafe logging of potentially sensitive data
2025-06-18T15:45:33.580+10:00  INFO 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Completed initialization in 3 ms
2025-06-18T15:45:33.625+10:00 DEBUG 10816 --- [           main] o.s.c.support.DefaultLifecycleProcessor  : Starting beans in phase -2147483647
2025-06-18T15:45:33.625+10:00 DEBUG 10816 --- [           main] o.s.c.support.DefaultLifecycleProcessor  : Successfully started bean 'springBootLoggingLifecycle'
2025-06-18T15:45:33.625+10:00 DEBUG 10816 --- [           main] o.s.c.support.DefaultLifecycleProcessor  : Starting beans in phase 2147483647
2025-06-18T15:45:33.640+10:00  INFO 10816 --- [           main] code.restful.AddressBookControllerTest   : Started AddressBookControllerTest in 3.777 seconds (process running for 5.105)
2025-06-18T15:45:33.648+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 0, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.652+10:00 DEBUG 10816 --- [           main] DependencyInjectionTestExecutionListener : Performing dependency injection for test class code.restful.AddressBookControllerTest
2025-06-18T15:45:33.652+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 1, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.660+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 2, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.664+10:00 DEBUG 10816 --- [           main] sContextBeforeModesTestExecutionListener : Before test method: class [AddressBookControllerTest], method [addContact_WithValidData_ShouldCreateContact], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
2025-06-18T15:45:33.666+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 3, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.667+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 4, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.673+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 5, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.811+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : POST "/contacts", parameters={}
2025-06-18T15:45:33.832+10:00 DEBUG 10816 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped to ResourceHttpRequestHandler [classpath [META-INF/resources/], classpath [resources/], classpath [static/], classpath [public/], ServletContext [/]]
2025-06-18T15:45:33.851+10:00 DEBUG 10816 --- [           main] o.s.w.s.r.ResourceHttpRequestHandler     : Resource not found
2025-06-18T15:45:33.857+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Using @ExceptionHandler code.restful.exception.GlobalExceptionHandler#handleAllUncaughtException(Exception)
2025-06-18T15:45:33.873+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/json', given [*/*] and supported [application/json, application/*+json]
2025-06-18T15:45:33.873+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Writing [{error=An unexpected error occurred}]
2025-06-18T15:45:33.886+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Resolved [org.springframework.web.servlet.resource.NoResourceFoundException: No static resource contacts.]
2025-06-18T15:45:33.888+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Completed 500 INTERNAL_SERVER_ERROR
2025-06-18T15:45:33.911+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 6, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.911+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 7, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.911+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 8, missCount = 1, failureCount = 0]

MockHttpServletRequest:
      HTTP Method = POST
      Request URI = /contacts
       Parameters = {}
          Headers = [Content-Type:"application/json;charset=UTF-8", Content-Length:"117"]
             Body = {"id":1,"firstName":"John","lastName":"Doe","phoneNumber":"1234567890","email":"john.doe@example.com","address":null}
    Session Attrs = {}

Handler:
             Type = org.springframework.web.servlet.resource.ResourceHttpRequestHandler

Async:
    Async started = false
     Async result = null

Resolved Exception:
             Type = org.springframework.web.servlet.resource.NoResourceFoundException

ModelAndView:
        View name = null
             View = null
            Model = null

FlashMap:
       Attributes = null

MockHttpServletResponse:
           Status = 500
    Error message = null
          Headers = [Vary:"Origin", "Access-Control-Request-Method", "Access-Control-Request-Headers", Content-Type:"application/json"]
     Content type = application/json
             Body = {"error":"An unexpected error occurred"}
    Forwarded URL = null
   Redirected URL = null
          Cookies = []
2025-06-18T15:45:33.911+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 9, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.911+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 10, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.911+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 11, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.911+10:00 DEBUG 10816 --- [           main] .c.s.DirtiesContextTestExecutionListener : After test method: class [AddressBookControllerTest], method [addContact_WithValidData_ShouldCreateContact], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
]]></system-out>
    <system-err><![CDATA[WARNING: A Java agent has been loaded dynamically (C:\Users\navee\.m2\repository\net\bytebuddy\byte-buddy-agent\1.14.12\byte-buddy-agent-1.14.12.jar)
WARNING: If a serviceability tool is in use, please run with -XX:+EnableDynamicAgentLoading to hide this warning
WARNING: If a serviceability tool is not in use, please run with -Djdk.instrument.traceUsage for more information
WARNING: Dynamic loading of agents will be disallowed by default in a future release
]]></system-err>
  </testcase>
  <testcase name="getContacts_WithSearchParams_ShouldReturnFilteredContacts" classname="code.restful.AddressBookControllerTest" time="0.018">
    <failure message="Status expected:&lt;200&gt; but was:&lt;500&gt;" type="java.lang.AssertionError"><![CDATA[java.lang.AssertionError: Status expected:<200> but was:<500>
	at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
	at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
	at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
	at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
	at code.restful.AddressBookControllerTest.getContacts_WithSearchParams_ShouldReturnFilteredContacts(AddressBookControllerTest.java:75)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
]]></failure>
    <system-out><![CDATA[2025-06-18T15:45:33.923+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 12, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.923+10:00 DEBUG 10816 --- [           main] DependencyInjectionTestExecutionListener : Performing dependency injection for test class code.restful.AddressBookControllerTest
2025-06-18T15:45:33.923+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 13, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.932+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 14, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.932+10:00 DEBUG 10816 --- [           main] sContextBeforeModesTestExecutionListener : Before test method: class [AddressBookControllerTest], method [getContacts_WithSearchParams_ShouldReturnFilteredContacts], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
2025-06-18T15:45:33.932+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 15, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.932+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 16, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.932+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 17, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.941+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : GET "/contacts", parameters={masked}
2025-06-18T15:45:33.942+10:00 DEBUG 10816 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped to ResourceHttpRequestHandler [classpath [META-INF/resources/], classpath [resources/], classpath [static/], classpath [public/], ServletContext [/]]
2025-06-18T15:45:33.946+10:00 DEBUG 10816 --- [           main] o.s.w.s.r.ResourceHttpRequestHandler     : Resource not found
2025-06-18T15:45:33.948+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Using @ExceptionHandler code.restful.exception.GlobalExceptionHandler#handleAllUncaughtException(Exception)
2025-06-18T15:45:33.948+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/json', given [*/*] and supported [application/json, application/*+json]
2025-06-18T15:45:33.948+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Writing [{error=An unexpected error occurred}]
2025-06-18T15:45:33.948+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Resolved [org.springframework.web.servlet.resource.NoResourceFoundException: No static resource contacts.]
2025-06-18T15:45:33.950+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Completed 500 INTERNAL_SERVER_ERROR
2025-06-18T15:45:33.950+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 18, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.950+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 19, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.950+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 20, missCount = 1, failureCount = 0]

MockHttpServletRequest:
      HTTP Method = GET
      Request URI = /contacts
       Parameters = {firstName=[John], lastName=[Doe]}
          Headers = []
             Body = null
    Session Attrs = {}

Handler:
             Type = org.springframework.web.servlet.resource.ResourceHttpRequestHandler

Async:
    Async started = false
     Async result = null

Resolved Exception:
             Type = org.springframework.web.servlet.resource.NoResourceFoundException

ModelAndView:
        View name = null
             View = null
            Model = null

FlashMap:
       Attributes = null

MockHttpServletResponse:
           Status = 500
    Error message = null
          Headers = [Vary:"Origin", "Access-Control-Request-Method", "Access-Control-Request-Headers", Content-Type:"application/json"]
     Content type = application/json
             Body = {"error":"An unexpected error occurred"}
    Forwarded URL = null
   Redirected URL = null
          Cookies = []
2025-06-18T15:45:33.950+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 21, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.950+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 22, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.950+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 23, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.950+10:00 DEBUG 10816 --- [           main] .c.s.DirtiesContextTestExecutionListener : After test method: class [AddressBookControllerTest], method [getContacts_WithSearchParams_ShouldReturnFilteredContacts], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
]]></system-out>
  </testcase>
  <testcase name="getContacts_ShouldReturnAllContacts" classname="code.restful.AddressBookControllerTest" time="0.016">
    <failure message="Status expected:&lt;200&gt; but was:&lt;500&gt;" type="java.lang.AssertionError"><![CDATA[java.lang.AssertionError: Status expected:<200> but was:<500>
	at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
	at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
	at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
	at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
	at code.restful.AddressBookControllerTest.getContacts_ShouldReturnAllContacts(AddressBookControllerTest.java:63)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
]]></failure>
    <system-out><![CDATA[2025-06-18T15:45:33.958+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 24, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.960+10:00 DEBUG 10816 --- [           main] DependencyInjectionTestExecutionListener : Performing dependency injection for test class code.restful.AddressBookControllerTest
2025-06-18T15:45:33.961+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 25, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.961+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 26, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.961+10:00 DEBUG 10816 --- [           main] sContextBeforeModesTestExecutionListener : Before test method: class [AddressBookControllerTest], method [getContacts_ShouldReturnAllContacts], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
2025-06-18T15:45:33.961+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 27, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.961+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 28, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.961+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 29, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.961+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : GET "/contacts", parameters={}
2025-06-18T15:45:33.961+10:00 DEBUG 10816 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped to ResourceHttpRequestHandler [classpath [META-INF/resources/], classpath [resources/], classpath [static/], classpath [public/], ServletContext [/]]
2025-06-18T15:45:33.961+10:00 DEBUG 10816 --- [           main] o.s.w.s.r.ResourceHttpRequestHandler     : Resource not found
2025-06-18T15:45:33.961+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Using @ExceptionHandler code.restful.exception.GlobalExceptionHandler#handleAllUncaughtException(Exception)
2025-06-18T15:45:33.973+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/json', given [*/*] and supported [application/json, application/*+json]
2025-06-18T15:45:33.973+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Writing [{error=An unexpected error occurred}]
2025-06-18T15:45:33.974+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Resolved [org.springframework.web.servlet.resource.NoResourceFoundException: No static resource contacts.]
2025-06-18T15:45:33.974+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Completed 500 INTERNAL_SERVER_ERROR
2025-06-18T15:45:33.976+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 30, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 31, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 32, missCount = 1, failureCount = 0]

MockHttpServletRequest:
      HTTP Method = GET
      Request URI = /contacts
       Parameters = {}
          Headers = []
             Body = null
    Session Attrs = {}

Handler:
             Type = org.springframework.web.servlet.resource.ResourceHttpRequestHandler

Async:
    Async started = false
     Async result = null

Resolved Exception:
             Type = org.springframework.web.servlet.resource.NoResourceFoundException

ModelAndView:
        View name = null
             View = null
            Model = null

FlashMap:
       Attributes = null

MockHttpServletResponse:
           Status = 500
    Error message = null
          Headers = [Vary:"Origin", "Access-Control-Request-Method", "Access-Control-Request-Headers", Content-Type:"application/json"]
     Content type = application/json
             Body = {"error":"An unexpected error occurred"}
    Forwarded URL = null
   Redirected URL = null
          Cookies = []
2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 33, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 34, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 35, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] .c.s.DirtiesContextTestExecutionListener : After test method: class [AddressBookControllerTest], method [getContacts_ShouldReturnAllContacts], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
]]></system-out>
  </testcase>
  <testcase name="deleteContact_WithInvalidId_ShouldReturnBadRequest" classname="code.restful.AddressBookControllerTest" time="0.031">
    <failure message="Status expected:&lt;400&gt; but was:&lt;500&gt;" type="java.lang.AssertionError"><![CDATA[java.lang.AssertionError: Status expected:<400> but was:<500>
	at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
	at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
	at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
	at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
	at code.restful.AddressBookControllerTest.deleteContact_WithInvalidId_ShouldReturnBadRequest(AddressBookControllerTest.java:139)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
]]></failure>
    <system-out><![CDATA[2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 36, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] DependencyInjectionTestExecutionListener : Performing dependency injection for test class code.restful.AddressBookControllerTest
2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 37, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 38, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] sContextBeforeModesTestExecutionListener : Before test method: class [AddressBookControllerTest], method [deleteContact_WithInvalidId_ShouldReturnBadRequest], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 39, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.977+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 40, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.988+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 41, missCount = 1, failureCount = 0]
2025-06-18T15:45:33.995+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : DELETE "/contacts/999", parameters={}
2025-06-18T15:45:33.995+10:00 DEBUG 10816 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped to ResourceHttpRequestHandler [classpath [META-INF/resources/], classpath [resources/], classpath [static/], classpath [public/], ServletContext [/]]
2025-06-18T15:45:33.997+10:00 DEBUG 10816 --- [           main] o.s.w.s.r.ResourceHttpRequestHandler     : Resource not found
2025-06-18T15:45:33.998+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Using @ExceptionHandler code.restful.exception.GlobalExceptionHandler#handleAllUncaughtException(Exception)
2025-06-18T15:45:33.999+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/json', given [*/*] and supported [application/json, application/*+json]
2025-06-18T15:45:33.999+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Writing [{error=An unexpected error occurred}]
2025-06-18T15:45:33.999+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Resolved [org.springframework.web.servlet.resource.NoResourceFoundException: No static resource contacts/999.]
2025-06-18T15:45:33.999+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Completed 500 INTERNAL_SERVER_ERROR
2025-06-18T15:45:34.001+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 42, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.001+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 43, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.001+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 44, missCount = 1, failureCount = 0]

MockHttpServletRequest:
      HTTP Method = DELETE
      Request URI = /contacts/999
       Parameters = {}
          Headers = []
             Body = null
    Session Attrs = {}

Handler:
             Type = org.springframework.web.servlet.resource.ResourceHttpRequestHandler

Async:
    Async started = false
     Async result = null

Resolved Exception:
             Type = org.springframework.web.servlet.resource.NoResourceFoundException

ModelAndView:
        View name = null
             View = null
            Model = null

FlashMap:
       Attributes = null

MockHttpServletResponse:
           Status = 500
    Error message = null
          Headers = [Vary:"Origin", "Access-Control-Request-Method", "Access-Control-Request-Headers", Content-Type:"application/json"]
     Content type = application/json
             Body = {"error":"An unexpected error occurred"}
    Forwarded URL = null
   Redirected URL = null
          Cookies = []
2025-06-18T15:45:34.002+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 45, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.003+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 46, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.004+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 47, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.004+10:00 DEBUG 10816 --- [           main] .c.s.DirtiesContextTestExecutionListener : After test method: class [AddressBookControllerTest], method [deleteContact_WithInvalidId_ShouldReturnBadRequest], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
]]></system-out>
  </testcase>
  <testcase name="getContact_WithInvalidId_ShouldReturnNotFound" classname="code.restful.AddressBookControllerTest" time="0.021">
    <failure message="Status expected:&lt;404&gt; but was:&lt;500&gt;" type="java.lang.AssertionError"><![CDATA[java.lang.AssertionError: Status expected:<404> but was:<500>
	at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
	at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
	at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
	at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
	at code.restful.AddressBookControllerTest.getContact_WithInvalidId_ShouldReturnNotFound(AddressBookControllerTest.java:93)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
]]></failure>
    <system-out><![CDATA[2025-06-18T15:45:34.011+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 48, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.011+10:00 DEBUG 10816 --- [           main] DependencyInjectionTestExecutionListener : Performing dependency injection for test class code.restful.AddressBookControllerTest
2025-06-18T15:45:34.011+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 49, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.011+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 50, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.011+10:00 DEBUG 10816 --- [           main] sContextBeforeModesTestExecutionListener : Before test method: class [AddressBookControllerTest], method [getContact_WithInvalidId_ShouldReturnNotFound], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
2025-06-18T15:45:34.019+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 51, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.019+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 52, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.019+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 53, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.023+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : GET "/contacts/999", parameters={}
2025-06-18T15:45:34.024+10:00 DEBUG 10816 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped to ResourceHttpRequestHandler [classpath [META-INF/resources/], classpath [resources/], classpath [static/], classpath [public/], ServletContext [/]]
2025-06-18T15:45:34.026+10:00 DEBUG 10816 --- [           main] o.s.w.s.r.ResourceHttpRequestHandler     : Resource not found
2025-06-18T15:45:34.026+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Using @ExceptionHandler code.restful.exception.GlobalExceptionHandler#handleAllUncaughtException(Exception)
2025-06-18T15:45:34.027+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/json', given [*/*] and supported [application/json, application/*+json]
2025-06-18T15:45:34.027+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Writing [{error=An unexpected error occurred}]
2025-06-18T15:45:34.028+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Resolved [org.springframework.web.servlet.resource.NoResourceFoundException: No static resource contacts/999.]
2025-06-18T15:45:34.028+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Completed 500 INTERNAL_SERVER_ERROR
2025-06-18T15:45:34.029+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 54, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.030+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 55, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.030+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 56, missCount = 1, failureCount = 0]

MockHttpServletRequest:
      HTTP Method = GET
      Request URI = /contacts/999
       Parameters = {}
          Headers = []
             Body = null
    Session Attrs = {}

Handler:
             Type = org.springframework.web.servlet.resource.ResourceHttpRequestHandler

Async:
    Async started = false
     Async result = null

Resolved Exception:
             Type = org.springframework.web.servlet.resource.NoResourceFoundException

ModelAndView:
        View name = null
             View = null
            Model = null

FlashMap:
       Attributes = null

MockHttpServletResponse:
           Status = 500
    Error message = null
          Headers = [Vary:"Origin", "Access-Control-Request-Method", "Access-Control-Request-Headers", Content-Type:"application/json"]
     Content type = application/json
             Body = {"error":"An unexpected error occurred"}
    Forwarded URL = null
   Redirected URL = null
          Cookies = []
2025-06-18T15:45:34.030+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 57, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.031+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 58, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.031+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 59, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.031+10:00 DEBUG 10816 --- [           main] .c.s.DirtiesContextTestExecutionListener : After test method: class [AddressBookControllerTest], method [getContact_WithInvalidId_ShouldReturnNotFound], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
]]></system-out>
  </testcase>
  <testcase name="getContact_WithValidId_ShouldReturnContact" classname="code.restful.AddressBookControllerTest" time="0.015">
    <failure message="Status expected:&lt;200&gt; but was:&lt;500&gt;" type="java.lang.AssertionError"><![CDATA[java.lang.AssertionError: Status expected:<200> but was:<500>
	at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
	at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
	at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
	at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
	at code.restful.AddressBookControllerTest.getContact_WithValidId_ShouldReturnContact(AddressBookControllerTest.java:84)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
]]></failure>
    <system-out><![CDATA[2025-06-18T15:45:34.035+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 60, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.036+10:00 DEBUG 10816 --- [           main] DependencyInjectionTestExecutionListener : Performing dependency injection for test class code.restful.AddressBookControllerTest
2025-06-18T15:45:34.036+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 61, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.040+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 62, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.043+10:00 DEBUG 10816 --- [           main] sContextBeforeModesTestExecutionListener : Before test method: class [AddressBookControllerTest], method [getContact_WithValidId_ShouldReturnContact], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
2025-06-18T15:45:34.044+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 63, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.044+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 64, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.045+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 65, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.047+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : GET "/contacts/1", parameters={}
2025-06-18T15:45:34.048+10:00 DEBUG 10816 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped to ResourceHttpRequestHandler [classpath [META-INF/resources/], classpath [resources/], classpath [static/], classpath [public/], ServletContext [/]]
2025-06-18T15:45:34.049+10:00 DEBUG 10816 --- [           main] o.s.w.s.r.ResourceHttpRequestHandler     : Resource not found
2025-06-18T15:45:34.049+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Using @ExceptionHandler code.restful.exception.GlobalExceptionHandler#handleAllUncaughtException(Exception)
2025-06-18T15:45:34.053+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/json', given [*/*] and supported [application/json, application/*+json]
2025-06-18T15:45:34.053+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Writing [{error=An unexpected error occurred}]
2025-06-18T15:45:34.053+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Resolved [org.springframework.web.servlet.resource.NoResourceFoundException: No static resource contacts/1.]
2025-06-18T15:45:34.053+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Completed 500 INTERNAL_SERVER_ERROR
2025-06-18T15:45:34.053+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 66, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.053+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 67, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.053+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 68, missCount = 1, failureCount = 0]

MockHttpServletRequest:
      HTTP Method = GET
      Request URI = /contacts/1
       Parameters = {}
          Headers = []
             Body = null
    Session Attrs = {}

Handler:
             Type = org.springframework.web.servlet.resource.ResourceHttpRequestHandler

Async:
    Async started = false
     Async result = null

Resolved Exception:
             Type = org.springframework.web.servlet.resource.NoResourceFoundException

ModelAndView:
        View name = null
             View = null
            Model = null

FlashMap:
       Attributes = null

MockHttpServletResponse:
           Status = 500
    Error message = null
          Headers = [Vary:"Origin", "Access-Control-Request-Method", "Access-Control-Request-Headers", Content-Type:"application/json"]
     Content type = application/json
             Body = {"error":"An unexpected error occurred"}
    Forwarded URL = null
   Redirected URL = null
          Cookies = []
2025-06-18T15:45:34.053+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 69, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.053+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 70, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.056+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 71, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.056+10:00 DEBUG 10816 --- [           main] .c.s.DirtiesContextTestExecutionListener : After test method: class [AddressBookControllerTest], method [getContact_WithValidId_ShouldReturnContact], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
]]></system-out>
  </testcase>
  <testcase name="updateContact_WithMismatchedId_ShouldReturnBadRequest" classname="code.restful.AddressBookControllerTest" time="0.018">
    <failure message="Status expected:&lt;400&gt; but was:&lt;500&gt;" type="java.lang.AssertionError"><![CDATA[java.lang.AssertionError: Status expected:<400> but was:<500>
	at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
	at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
	at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
	at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
	at code.restful.AddressBookControllerTest.updateContact_WithMismatchedId_ShouldReturnBadRequest(AddressBookControllerTest.java:125)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
]]></failure>
    <system-out><![CDATA[2025-06-18T15:45:34.059+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 72, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.060+10:00 DEBUG 10816 --- [           main] DependencyInjectionTestExecutionListener : Performing dependency injection for test class code.restful.AddressBookControllerTest
2025-06-18T15:45:34.060+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 73, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.060+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 74, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.062+10:00 DEBUG 10816 --- [           main] sContextBeforeModesTestExecutionListener : Before test method: class [AddressBookControllerTest], method [updateContact_WithMismatchedId_ShouldReturnBadRequest], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
2025-06-18T15:45:34.062+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 75, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.063+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 76, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.064+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 77, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.067+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : PUT "/contacts/2", parameters={}
2025-06-18T15:45:34.067+10:00 DEBUG 10816 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped to ResourceHttpRequestHandler [classpath [META-INF/resources/], classpath [resources/], classpath [static/], classpath [public/], ServletContext [/]]
2025-06-18T15:45:34.070+10:00 DEBUG 10816 --- [           main] o.s.w.s.r.ResourceHttpRequestHandler     : Resource not found
2025-06-18T15:45:34.070+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Using @ExceptionHandler code.restful.exception.GlobalExceptionHandler#handleAllUncaughtException(Exception)
2025-06-18T15:45:34.070+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/json', given [*/*] and supported [application/json, application/*+json]
2025-06-18T15:45:34.070+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Writing [{error=An unexpected error occurred}]
2025-06-18T15:45:34.072+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Resolved [org.springframework.web.servlet.resource.NoResourceFoundException: No static resource contacts/2.]
2025-06-18T15:45:34.072+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Completed 500 INTERNAL_SERVER_ERROR
2025-06-18T15:45:34.075+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 78, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.075+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 79, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.076+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 80, missCount = 1, failureCount = 0]

MockHttpServletRequest:
      HTTP Method = PUT
      Request URI = /contacts/2
       Parameters = {}
          Headers = [Content-Type:"application/json;charset=UTF-8", Content-Length:"117"]
             Body = {"id":1,"firstName":"John","lastName":"Doe","phoneNumber":"1234567890","email":"john.doe@example.com","address":null}
    Session Attrs = {}

Handler:
             Type = org.springframework.web.servlet.resource.ResourceHttpRequestHandler

Async:
    Async started = false
     Async result = null

Resolved Exception:
             Type = org.springframework.web.servlet.resource.NoResourceFoundException

ModelAndView:
        View name = null
             View = null
            Model = null

FlashMap:
       Attributes = null

MockHttpServletResponse:
           Status = 500
    Error message = null
          Headers = [Vary:"Origin", "Access-Control-Request-Method", "Access-Control-Request-Headers", Content-Type:"application/json"]
     Content type = application/json
             Body = {"error":"An unexpected error occurred"}
    Forwarded URL = null
   Redirected URL = null
          Cookies = []
2025-06-18T15:45:34.077+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 81, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.077+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 82, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.078+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 83, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.078+10:00 DEBUG 10816 --- [           main] .c.s.DirtiesContextTestExecutionListener : After test method: class [AddressBookControllerTest], method [updateContact_WithMismatchedId_ShouldReturnBadRequest], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
]]></system-out>
  </testcase>
  <testcase name="updateContact_WithValidData_ShouldUpdateContact" classname="code.restful.AddressBookControllerTest" time="0.013">
    <failure message="Status expected:&lt;200&gt; but was:&lt;500&gt;" type="java.lang.AssertionError"><![CDATA[java.lang.AssertionError: Status expected:<200> but was:<500>
	at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
	at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
	at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
	at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
	at code.restful.AddressBookControllerTest.updateContact_WithValidData_ShouldUpdateContact(AddressBookControllerTest.java:115)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
]]></failure>
    <system-out><![CDATA[2025-06-18T15:45:34.081+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 84, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.082+10:00 DEBUG 10816 --- [           main] DependencyInjectionTestExecutionListener : Performing dependency injection for test class code.restful.AddressBookControllerTest
2025-06-18T15:45:34.082+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 85, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.083+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 86, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.084+10:00 DEBUG 10816 --- [           main] sContextBeforeModesTestExecutionListener : Before test method: class [AddressBookControllerTest], method [updateContact_WithValidData_ShouldUpdateContact], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
2025-06-18T15:45:34.084+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 87, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.084+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 88, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.086+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 89, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.089+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : PUT "/contacts/1", parameters={}
2025-06-18T15:45:34.090+10:00 DEBUG 10816 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped to ResourceHttpRequestHandler [classpath [META-INF/resources/], classpath [resources/], classpath [static/], classpath [public/], ServletContext [/]]
2025-06-18T15:45:34.092+10:00 DEBUG 10816 --- [           main] o.s.w.s.r.ResourceHttpRequestHandler     : Resource not found
2025-06-18T15:45:34.092+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Using @ExceptionHandler code.restful.exception.GlobalExceptionHandler#handleAllUncaughtException(Exception)
2025-06-18T15:45:34.093+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/json', given [*/*] and supported [application/json, application/*+json]
2025-06-18T15:45:34.093+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Writing [{error=An unexpected error occurred}]
2025-06-18T15:45:34.094+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Resolved [org.springframework.web.servlet.resource.NoResourceFoundException: No static resource contacts/1.]
2025-06-18T15:45:34.094+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Completed 500 INTERNAL_SERVER_ERROR
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 90, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 91, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 92, missCount = 1, failureCount = 0]

MockHttpServletRequest:
      HTTP Method = PUT
      Request URI = /contacts/1
       Parameters = {}
          Headers = [Content-Type:"application/json;charset=UTF-8", Content-Length:"117"]
             Body = {"id":1,"firstName":"John","lastName":"Doe","phoneNumber":"1234567890","email":"john.doe@example.com","address":null}
    Session Attrs = {}

Handler:
             Type = org.springframework.web.servlet.resource.ResourceHttpRequestHandler

Async:
    Async started = false
     Async result = null

Resolved Exception:
             Type = org.springframework.web.servlet.resource.NoResourceFoundException

ModelAndView:
        View name = null
             View = null
            Model = null

FlashMap:
       Attributes = null

MockHttpServletResponse:
           Status = 500
    Error message = null
          Headers = [Vary:"Origin", "Access-Control-Request-Method", "Access-Control-Request-Headers", Content-Type:"application/json"]
     Content type = application/json
             Body = {"error":"An unexpected error occurred"}
    Forwarded URL = null
   Redirected URL = null
          Cookies = []
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 93, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 94, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 95, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] .c.s.DirtiesContextTestExecutionListener : After test method: class [AddressBookControllerTest], method [updateContact_WithValidData_ShouldUpdateContact], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
]]></system-out>
  </testcase>
  <testcase name="deleteContact_WithValidId_ShouldDeleteContact" classname="code.restful.AddressBookControllerTest" time="0.016">
    <failure message="Status expected:&lt;204&gt; but was:&lt;500&gt;" type="java.lang.AssertionError"><![CDATA[java.lang.AssertionError: Status expected:<204> but was:<500>
	at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
	at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
	at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
	at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
	at code.restful.AddressBookControllerTest.deleteContact_WithValidId_ShouldDeleteContact(AddressBookControllerTest.java:132)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
]]></failure>
    <system-out><![CDATA[2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 96, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] DependencyInjectionTestExecutionListener : Performing dependency injection for test class code.restful.AddressBookControllerTest
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 97, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 98, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] sContextBeforeModesTestExecutionListener : Before test method: class [AddressBookControllerTest], method [deleteContact_WithValidId_ShouldDeleteContact], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 99, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.096+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 100, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.106+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 101, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.109+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : DELETE "/contacts/1", parameters={}
2025-06-18T15:45:34.109+10:00 DEBUG 10816 --- [           main] o.s.w.s.handler.SimpleUrlHandlerMapping  : Mapped to ResourceHttpRequestHandler [classpath [META-INF/resources/], classpath [resources/], classpath [static/], classpath [public/], ServletContext [/]]
2025-06-18T15:45:34.110+10:00 DEBUG 10816 --- [           main] o.s.w.s.r.ResourceHttpRequestHandler     : Resource not found
2025-06-18T15:45:34.110+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Using @ExceptionHandler code.restful.exception.GlobalExceptionHandler#handleAllUncaughtException(Exception)
2025-06-18T15:45:34.112+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/json', given [*/*] and supported [application/json, application/*+json]
2025-06-18T15:45:34.112+10:00 DEBUG 10816 --- [           main] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Writing [{error=An unexpected error occurred}]
2025-06-18T15:45:34.112+10:00 DEBUG 10816 --- [           main] .m.m.a.ExceptionHandlerExceptionResolver : Resolved [org.springframework.web.servlet.resource.NoResourceFoundException: No static resource contacts/1.]
2025-06-18T15:45:34.112+10:00 DEBUG 10816 --- [           main] o.s.t.web.servlet.TestDispatcherServlet  : Completed 500 INTERNAL_SERVER_ERROR
2025-06-18T15:45:34.112+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 102, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.112+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 103, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.112+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 104, missCount = 1, failureCount = 0]

MockHttpServletRequest:
      HTTP Method = DELETE
      Request URI = /contacts/1
       Parameters = {}
          Headers = []
             Body = null
    Session Attrs = {}

Handler:
             Type = org.springframework.web.servlet.resource.ResourceHttpRequestHandler

Async:
    Async started = false
     Async result = null

Resolved Exception:
             Type = org.springframework.web.servlet.resource.NoResourceFoundException

ModelAndView:
        View name = null
             View = null
            Model = null

FlashMap:
       Attributes = null

MockHttpServletResponse:
           Status = 500
    Error message = null
          Headers = [Vary:"Origin", "Access-Control-Request-Method", "Access-Control-Request-Headers", Content-Type:"application/json"]
     Content type = application/json
             Body = {"error":"An unexpected error occurred"}
    Forwarded URL = null
   Redirected URL = null
          Cookies = []
2025-06-18T15:45:34.112+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 105, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.112+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 106, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.112+10:00 DEBUG 10816 --- [           main] org.springframework.test.context.cache   : Spring test ApplicationContext cache statistics: [DefaultContextCache@38c460e8 size = 1, maxSize = 32, parentContextCount = 0, hitCount = 107, missCount = 1, failureCount = 0]
2025-06-18T15:45:34.112+10:00 DEBUG 10816 --- [           main] .c.s.DirtiesContextTestExecutionListener : After test method: class [AddressBookControllerTest], method [deleteContact_WithValidId_ShouldDeleteContact], class annotated with @DirtiesContext [false] with mode [null], method annotated with @DirtiesContext [false] with mode [null]
]]></system-out>
  </testcase>
</testsuite>